lib/thrift/exceptions.ex

```diff
@@ -14,3 +14,19 @@ defmodule Thrift.TApplicationException do
   def exception_type(10), do: :unsupported_client_type
   def exception_type(_), do: :unknown
 end
+
+defmodule Thrift.Union.TooManyFieldsSetException do
+  @moduledoc """
+  This exception occurs when a Union is serialized and more than one
+  field is set.
+  """
+  defexception message: nil, set_fields: nil
+
+  def new(set_fields) do
+    field_names = Enum.map(set_fields, &elem(&1, 0))
+    %__MODULE__{
+      message: "Thrift union has more than one field set",
+      set_fields: field_names
+    }
+  end
```

Comment:
This method is only used in one place. It also has high coupling with that location in the way it takes set_fields but doesn't use all the tuple elements. Recommend inlining it.